src/lib/structures/Command.ts

```diff
@@ -0,0 +1,214 @@
+import { AliasPiece, AliasPieceOptions, Permissions, PermissionsResolvable } from '@klasa/core';
+
+import type { CommandStore } from './CommandStore';
+import type { ChannelType } from '@klasa/dapi-types';
+
+import { RateLimitManager } from '@klasa/ratelimits';
+import { mergeDefault } from '@klasa/utils';
+
+import { commandDefaults } from '../util/constants';
+import { CooldownLevel } from '../types/Enums';
+
+export abstract class Command extends AliasPiece {
+	/**
+	 * Permissions required by the bot to run the command
+	 * @since 0.0.1
+	 */
+	public requiredPermissions: Permissions;
+
+	/**
+	 * Delete command's response if the trigger message was deleted
+	 * @since 0.0.1
+	 */
+	public deletable: boolean;
+
+	/**
+	 * A basic summary about the command
+	 * @since 0.0.1
+	 */
+	public description: string;
+
+	/**
+	 * Longer version of command's summary and how to use it
+	 * @since 0.0.1
+	 */
+	public extendedHelp: string;
+
+	/**
+	 * Full category name of the command
+	 * @since 0.0.1
+	 */
+	public fullCategory: string[];
+
+	/**
+	 * Allow disabling of the command in a guild or not
+	 * @since 0.0.1
+	 */
+	public guarded: boolean;
```

Comment:
this can also use a better name

Reply:
This made the most sense for its purpose.